### PR TITLE
Add lora_path to chat completion

### PR DESCRIPTION
**Pull request by ccchow**
## Motivation
Add lora_path to  ChatCompletionRequest  for OpenAI chat completion API. It was previously added to OpenAI completion API https://github.com/sgl-project/sglang/pull/2243

## Modifications
Added lora_path to ChatCompletionRequest

## Checklist

- [x] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [ ] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [ ] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

File: python/sglang/srt/openai_api/protocol.py [hunk 1]
```diff
@@ -274,6 +274,7 @@ class ChatCompletionRequest(BaseModel):
     temperature: float = 0.7
     top_p: float = 1.0
     user: Optional[str] = None
+    lora_path: Optional[Union[List[Optional[str]], Optional[str]]] = None
 
     # Extra parameters for SRT backend only and will be ignored by OpenAI models.
     regex: Optional[str] = None
```
Comment (merrymercy) [inline python/sglang/srt/openai_api/protocol.py]:
```suggestion

    # Extra parameters for SRT backend only and will be ignored by OpenAI models.
    lora_path: Optional[Union[List[Optional[str]], Optional[str]]] = None
    regex: Optional[str] = None
```

**Comment by merrymercy** [comment]
Even if you added this, it seems it is still not used.

**Comment by ccchow** [comment]
We are working on a project to provide multi-lora serving via OpenAI compatible API, and we have validated this fix by adding lora_path to OpenAI protocol and serving a batch with different lora adapters.

**Comment by ccchow** [comment]
Thanks for merging this change!

**Comment by qingzhong1** [comment]
Hello, how to use url = "http://localhost:8000/v1/chat/completions" to request the configured lora, data = {"model": "Qwen2.5-7B-Instruct","messages": [{ "role": "user", "content": "What is the capital of France?"}]},lora name='aa'

**Comment by ccchow** [comment]
curl -X POST http://127.0.0.1:30000/v1/chat/completions -d '{"model": "meta-llama/Llama-3.2-1B", "messages": [{"role": "system", "content": "You are a happy assistant that puts a positive spin on everything."}, {"role": "user", "content": "   "}], "lora_path": "lora1"}'

**Comment by ccchow** [comment]
> Hello, how to use url = "http://localhost:8000/v1/chat/completions" to request the configured lora, data = {"model": "Qwen2.5-7B-Instruct","messages": [{ "role": "user", "content": "What is the capital of France?"}]},lora name='aa'

curl -X POST http://127.0.0.1:30000/v1/chat/completions -d '{"model": "meta-llama/Llama-3.2-1B", "messages": [{"role": "system", "content": "You are a happy assistant that puts a positive spin on everything."}, {"role": "user", "content": "I fell off my bike today."}], "lora_path": "lora1", "max_tokens": 64}'

**Comment by qingzhong1** [comment]
v1_completions can successfully call lora, but v1/chat/completions cannot call lora, why?Comparing v1_generate_request and v1_chat_generate_request, we found that v1_chat_generate_request does not have the lora_pathes variable

**Comment by ccchow** [comment]
> v1_completions can successfully call lora, but v1/chat/completions cannot call lora, why?Comparing v1_generate_request and v1_chat_generate_request, we found that v1_chat_generate_request does not have the lora_pathes variable

You are right. I missed that when cherry picking changes. Will have another PR.

**Comment by ccchow** [comment]
> v1_completions can successfully call lora, but v1/chat/completions cannot call lora, why?Comparing v1_generate_request and v1_chat_generate_request, we found that v1_chat_generate_request does not have the lora_pathes variable

https://github.com/sgl-project/sglang/pull/2529

**Pull request merged**